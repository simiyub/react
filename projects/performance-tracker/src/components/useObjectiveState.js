import {React, useState} from 'react';

const useObjectiveState=()=>{
    const [state, setValues] = useState({'objective':''});

    const handleValueChange =(e)=>{
        setValues({[e.target.name]:e.target.value});
    }    

    return{
        state,
        handleValueChange
    }
        
    
    
}

export default useObjectiveState;