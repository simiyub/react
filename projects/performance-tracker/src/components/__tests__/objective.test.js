import React from 'react';
import { render } from '@testing-library/react';
import Objective from '../objective';
import {mount, shallow} from 'enzyme';
import { testHook } from './testHook';
import {useObjectiveState} from '../useObjectiveState';
let wrapper;
let stateUtil;

beforeEach(()=>{
    wrapper = shallow(<Objective/>);

    testHook(()=>{

        stateUtil = useObjectiveState();
    })

})
describe("Objective tests", ()=>{

    it("matchesthe snapshot", ()=>{
        expect(wrapper.html()).toMatchSnapshot();
    })
    it("renders without crashing",()=>{
        shallow(<Objective/>);
    })
    it("has required input", ()=>{
        expect(wrapper.find(".objective").length).toBe(1);
        expect(wrapper.find(".startdate").length).toBe(1);
        expect(wrapper.find(".enddate").length).toBe(1);
        expect(wrapper.find(".owner").length).toBe(1);
        expect(wrapper.find(".status").length).toBe(1);
        expect(wrapper.find(".logo").length).toBe(1);
    })
    it("has required buttons", ()=>{
        expect(wrapper.find(".submit").length).toBe(1);
        expect(wrapper.find(".reset").length).toBe(1);
    })
    
    it("calls a function passed to it when clicked",()=>{
        const mockCallBack = jest.fn();
        const mountedObjectiveWithCallback = shallow(<Objective handleClick = {mockCallBack}/>);
        mountedObjectiveWithCallback.find(".submit").simulate('click');
        expect(mockCallBack.mock.calls.length).toEqual(1);
    })

    it("calls setState with the value of objective provided", ()=>{
        let myFirstObjective="myfirst objective";
        const setState = jest.fn();
        const useStateSpy = jest.spyOn(React, 'useState')
        useStateSpy.mockImplementation((init)=>[init, setState]);
        wrapper.find('objective').onChange();
        expect(setState).toHaveBeenCalledWith(myFirstObjective);

        // const mountedObjective = mount(<Objective/>);
        // let myFirstObjective="myfirst objective";
        // //const mockCallBack = jest.fn(mountedObjective.handleValueChange);
        // const mockEvent = {target:{name:"objective", value:myFirstObjective}};
        // //expect(mountedObjective.find('input[type="password"'))
        // //mockCallBack(mockEvent);
        // const mockCallBack=jest.fn( mountedObjective.handleValueChange);
        // mockCallBack(mockEvent);
        // expect(mountedObjective.state.objective.toEqual(myFirstObjective))
    })

    it("calls axios.get in #componentDidMount", ()=>{
        return wrapper.instance().componentDidMount().then( ()=>{
            expect(axios.get).toHaveBeenCalled();
        })
    })

})