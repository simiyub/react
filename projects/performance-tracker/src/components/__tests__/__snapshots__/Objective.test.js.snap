// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Objective tests matchesthe snapshot 1`] = `"<form><img class=\\"logo\\" src=\\"./images/objective.png\\"/><h3>Create your objectives here</h3><div><label>Objective</label><textarea id=\\"objective\\" type=\\"text\\" name=\\"objective\\" class=\\"objective\\"></textarea></div><div><label>Start Date</label><input type=\\"date\\" class=\\"startdate\\"/></div><div><label>End Date</label><input type=\\"date\\" class=\\"enddate\\"/></div><div><label>Owner</label><select class=\\"owner\\"><option value=\\"bramwell\\">Bramwell</option></select></div><div><label>Status</label><select class=\\"status\\"><option value=\\"new\\">New</option><option value=\\"inprogress\\">In Progress</option><option value=\\"onhold\\">On Hold</option><option value=\\"done\\">Done</option></select></div><button class=\\"submit\\" type=\\"submit\\">Submit</button><button class=\\"reset\\" type=\\"reset\\">Reset</button></form>"`;
